### ISSUE TITLE
algorithm(2417): error C2678: binary '*': no operator found which takes a left-hand operand of type 'const Marks::const_iterator' (or there is no acceptable conversion)

### ISSUE THREAD

**Issue by evandrocoan** [variar/klogg#35]
Details:

<details>
<p>

```
Microsoft Windows [Version 10.0.15063]
(c) 2017 Microsoft Corporation. All rights reserved.

D:\glogg>"F:\Qt\5.10.0\msvc2017_64\bin\qtenv2.bat"
Setting up environment for Qt usage...
Remember to call vcvarsall.bat to complete environment setup!

F:\Qt\5.10.0\msvc2017_64>"F:\VisualStudio2017\VC\Auxiliary\Build\vcvars64.bat"
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.5.3
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

F:\Qt\5.10.0\msvc2017_64>cd /d D:\glogg

D:\glogg>md build
D:\glogg>cd build

D:\glogg\build>D:\User\Documents\CMake\bin\cmake.exe -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=Release -DBUILD_VERSION=X.X.X ../
Version override X.X.X
Build
Project version is 17.12.0
Project version tweak is 0

CMAKE_C_FLAGS: /DWIN32 /D_WINDOWS /W4 /O2 /wd4127
CMAKE_CXX_FLAGS: /DWIN32 /D_WINDOWS /W4 /GR /EHsc /O2 /wd4127
CMAKE_EXE_LINKER_FLAGS: /machine:x64
CMAKE_MODULE_LINKER_FLAGS: /machine:x64
CMAKE_SHARED_LINKER_FLAGS: /machine:x64
CMAKE_STATIC_LINKER_FLAGS: /machine:x64
CMAKE_C_FLAGS_RELEASE: /MD /O2 /Ob2 /DNDEBUG
CMAKE_CXX_FLAGS_RELEASE: /MD /O2 /Ob2 /DNDEBUG
CMAKE_EXE_LINKER_FLAGS_RELEASE: /INCREMENTAL:NO
CMAKE_MODULE_LINKER_FLAGS_RELEASE: /INCREMENTAL:NO
CMAKE_SHARED_LINKER_FLAGS_RELEASE: /INCREMENTAL:NO
CMAKE_STATIC_LINKER_FLAGS_RELEASE:

Support for cross-platform IPC will be included
CMake Error at cmake/pandocology.cmake:48 (message):
  Pandoc not found.  Install Pandoc (http://johnmacfarlane.net/pandoc/) or
  set cache variable PANDOC_EXECUTABLE.
Call Stack (most recent call first):
  CMakeLists.txt:164 (include)


-- Configuring incomplete, errors occurred!
See also "D:/User/Dropbox/SoftwareVersioning/glogg/build/CMakeFiles/CMakeOutput.log".

D:\glogg\build>D:\User\Documents\CMake\bin\cmake.exe -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=Release -DBUILD_VERSION=X.X.X ../
Version override X.X.X
Build
Project version is 17.12.0
Project version tweak is 0

CMAKE_C_FLAGS: /DWIN32 /D_WINDOWS /W4 /O2 /wd4127
CMAKE_CXX_FLAGS: /DWIN32 /D_WINDOWS /W4 /GR /EHsc /O2 /wd4127
CMAKE_EXE_LINKER_FLAGS: /machine:x64
CMAKE_MODULE_LINKER_FLAGS: /machine:x64
CMAKE_SHARED_LINKER_FLAGS: /machine:x64
CMAKE_STATIC_LINKER_FLAGS: /machine:x64
CMAKE_C_FLAGS_RELEASE: /MD /O2 /Ob2 /DNDEBUG
CMAKE_CXX_FLAGS_RELEASE: /MD /O2 /Ob2 /DNDEBUG
CMAKE_EXE_LINKER_FLAGS_RELEASE: /INCREMENTAL:NO
CMAKE_MODULE_LINKER_FLAGS_RELEASE: /INCREMENTAL:NO
CMAKE_SHARED_LINKER_FLAGS_RELEASE: /INCREMENTAL:NO
CMAKE_STATIC_LINKER_FLAGS_RELEASE:

Support for cross-platform IPC will be included
-- Configuring done
-- Generating done
-- Build files have been written to: D:/User/Dropbox/SoftwareVersioning/glogg/build

D:\glogg\build>"D:\User\Documents\CMake\bin\cmake.exe" --build . --config Release
Scanning dependencies of target generate_version
Git branch: master-variar
Git commit: d066dd8
Git commit: 17.12.0-2-gd066dd8
[  0%] Built target generate_version
Scanning dependencies of target html_readme
[  1%] Generating documentation.rst
[  1%] Generating readme.html
[  1%] Built target html_readme
Scanning dependencies of target uchardet_autogen
[  1%] Automatic MOC for target uchardet
Generating MOC compilation uchardet_autogen/moc_compilation.cpp
[  1%] Built target uchardet_autogen
Scanning dependencies of target uchardet
[  1%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/CharDistribution.cpp.obj
CharDistribution.cpp
[  2%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/JpCntx.cpp.obj
JpCntx.cpp
[  3%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangArabicModel.cpp.obj
LangArabicModel.cpp
[  3%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangBulgarianModel.cpp.obj
LangBulgarianModel.cpp
[  4%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangRussianModel.cpp.obj
LangRussianModel.cpp
[  5%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangEsperantoModel.cpp.obj
LangEsperantoModel.cpp
[  5%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangFrenchModel.cpp.obj
LangFrenchModel.cpp
[  6%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangDanishModel.cpp.obj
LangDanishModel.cpp
[  7%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangGermanModel.cpp.obj
LangGermanModel.cpp
[  8%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangGreekModel.cpp.obj
LangGreekModel.cpp
[  8%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangHungarianModel.cpp.obj
LangHungarianModel.cpp
[  9%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangHebrewModel.cpp.obj
LangHebrewModel.cpp
[ 10%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangSpanishModel.cpp.obj
LangSpanishModel.cpp
[ 10%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangThaiModel.cpp.obj
LangThaiModel.cpp
[ 11%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangTurkishModel.cpp.obj
LangTurkishModel.cpp
[ 12%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/LangModels/LangVietnameseModel.cpp.obj
LangVietnameseModel.cpp
[ 12%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsHebrewProber.cpp.obj
nsHebrewProber.cpp
[ 13%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsCharSetProber.cpp.obj
nsCharSetProber.cpp
D:\glogg\3rdparty\uchardet\src\nsCharSetProber.cpp(77): warning C4244: '=': conversion from '__int64' to 'PRUint32', possible loss of data
D:\glogg\3rdparty\uchardet\src\nsCharSetProber.cpp(122): warning C4244: '=': conversion from '__int64' to 'PRUint32', possible loss of data
[ 14%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsBig5Prober.cpp.obj
nsBig5Prober.cpp
[ 14%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsEUCJPProber.cpp.obj
nsEUCJPProber.cpp
[ 15%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsEUCKRProber.cpp.obj
nsEUCKRProber.cpp
[ 16%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsEUCTWProber.cpp.obj
nsEUCTWProber.cpp
[ 17%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsEscCharsetProber.cpp.obj
nsEscCharsetProber.cpp
[ 17%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsEscSM.cpp.obj
nsEscSM.cpp
[ 18%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsGB2312Prober.cpp.obj
nsGB2312Prober.cpp
[ 19%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsMBCSGroupProber.cpp.obj
nsMBCSGroupProber.cpp
[ 19%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsMBCSSM.cpp.obj
nsMBCSSM.cpp
[ 20%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsSBCSGroupProber.cpp.obj
nsSBCSGroupProber.cpp
[ 21%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsSBCharSetProber.cpp.obj
nsSBCharSetProber.cpp
[ 21%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsSJISProber.cpp.obj
nsSJISProber.cpp
[ 22%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsUTF8Prober.cpp.obj
nsUTF8Prober.cpp
[ 23%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsLatin1Prober.cpp.obj
nsLatin1Prober.cpp
[ 24%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/nsUniversalDetector.cpp.obj
nsUniversalDetector.cpp
[ 24%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/src/uchardet.cpp.obj
uchardet.cpp
[ 25%] Building CXX object 3rdparty/uchardet/CMakeFiles/uchardet.dir/uchardet_autogen/moc_compilation.cpp.obj
moc_compilation.cpp
[ 26%] Linking CXX static library ..\..\output\uchardet.lib
[ 26%] Built target uchardet
Scanning dependencies of target efsw_autogen
[ 27%] Automatic MOC for target efsw
Generating MOC compilation efsw_autogen/moc_compilation.cpp
[ 27%] Built target efsw_autogen
Scanning dependencies of target efsw
[ 27%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/Debug.cpp.obj
Debug.cpp
[ 28%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/DirectorySnapshot.cpp.obj
DirectorySnapshot.cpp
[ 29%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/DirectorySnapshotDiff.cpp.obj
DirectorySnapshotDiff.cpp
[ 29%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/DirWatcherGeneric.cpp.obj
DirWatcherGeneric.cpp
[ 30%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/FileInfo.cpp.obj
FileInfo.cpp
[ 31%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/FileSystem.cpp.obj
FileSystem.cpp
[ 31%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/FileWatcher.cpp.obj
FileWatcher.cpp
[ 32%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/FileWatcherCWrapper.cpp.obj
FileWatcherCWrapper.cpp
[ 33%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/FileWatcherGeneric.cpp.obj
FileWatcherGeneric.cpp
D:\glogg\3rdparty\efsw\src\efsw\FileWatcherGeneric.cpp(153): warning C4100: 'action': unreferenced formal parameter
D:\glogg\3rdparty\efsw\src\efsw\FileWatcherGeneric.cpp(153): warning C4100: 'filename': unreferenced formal parameter
D:\glogg\3rdparty\efsw\src\efsw\FileWatcherGeneric.cpp(153): warning C4100: 'watch': unreferenced formal parameter
[ 33%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/FileWatcherImpl.cpp.obj
FileWatcherImpl.cpp
[ 34%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/Log.cpp.obj
Log.cpp
[ 35%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/Mutex.cpp.obj
Mutex.cpp
[ 36%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/String.cpp.obj
String.cpp
D:\glogg\3rdparty\efsw\src\efsw\String.cpp(473): warning C4100: 'n': unreferenced formal parameter
D:\glogg\3rdparty\efsw\src\efsw\String.cpp(512): warning C4100: 'n': unreferenced formal parameter
D:\glogg\3rdparty\efsw\src\efsw\String.cpp(620): warning C4100: 'n': unreferenced formal parameter
D:\glogg\3rdparty\efsw\src\efsw\String.cpp(640): warning C4100: 'n': unreferenced formal parameter
D:\glogg\3rdparty\efsw\src\efsw\String.cpp(700): warning C4100: 'n': unreferenced formal parameter
D:\glogg\3rdparty\efsw\src\efsw\String.cpp(720): warning C4100: 'n': unreferenced formal parameter
D:\glogg\3rdparty\efsw\src\efsw\String.cpp(740): warning C4100: 'n': unreferenced formal parameter
D:\glogg\3rdparty\efsw\src\efsw\String.cpp(760): warning C4100: 'n': unreferenced formal parameter
[ 36%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/System.cpp.obj
System.cpp
[ 37%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/Thread.cpp.obj
Thread.cpp
[ 38%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/Watcher.cpp.obj
Watcher.cpp
[ 38%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/WatcherGeneric.cpp.obj
WatcherGeneric.cpp
[ 39%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/platform/win/FileSystemImpl.cpp.obj
FileSystemImpl.cpp
[ 40%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/platform/win/MutexImpl.cpp.obj
MutexImpl.cpp
[ 40%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/platform/win/SystemImpl.cpp.obj
SystemImpl.cpp
[ 41%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/platform/win/ThreadImpl.cpp.obj
ThreadImpl.cpp
[ 42%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/FileWatcherWin32.cpp.obj
FileWatcherWin32.cpp
d:\glogg\3rdparty\efsw\src\efsw\filewatcherwin32.cpp(281) : warning C4701: potentially uninitialized local variable 'fwAction' used
[ 42%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/src/efsw/WatcherWin32.cpp.obj
WatcherWin32.cpp
[ 43%] Building CXX object 3rdparty/efsw/CMakeFiles/efsw.dir/efsw_autogen/moc_compilation.cpp.obj
moc_compilation.cpp
[ 44%] Linking CXX static library ..\..\output\efsw.lib
[ 44%] Built target efsw
Scanning dependencies of target utils_autogen
[ 45%] Automatic MOC for target utils
Generating MOC compilation utils_autogen/moc_compilation.cpp
[ 45%] Built target utils_autogen
Scanning dependencies of target utils
[ 46%] Building CXX object src/utils/CMakeFiles/utils.dir/src/persistentinfo.cpp.obj
persistentinfo.cpp
[ 46%] Building CXX object src/utils/CMakeFiles/utils.dir/src/configuration.cpp.obj
configuration.cpp
[ 47%] Building CXX object src/utils/CMakeFiles/utils.dir/utils_autogen/moc_compilation.cpp.obj
moc_compilation.cpp
[ 48%] Linking CXX static library ..\..\output\utils.lib
[ 48%] Built target utils
Scanning dependencies of target filewatch_autogen
[ 49%] Automatic MOC for target filewatch
Generating MOC source filewatch_autogen/6YEA5652QU/moc_filewatcher.cpp
Generating MOC compilation filewatch_autogen/moc_compilation.cpp
[ 49%] Built target filewatch_autogen
Scanning dependencies of target filewatch
[ 50%] Building CXX object src/filewatch/CMakeFiles/filewatch.dir/src/filewatcher.cpp.obj
filewatcher.cpp
[ 51%] Building CXX object src/filewatch/CMakeFiles/filewatch.dir/filewatch_autogen/moc_compilation.cpp.obj
moc_compilation.cpp
[ 51%] Linking CXX static library ..\..\output\filewatch.lib
[ 51%] Built target filewatch
Scanning dependencies of target logdata_autogen
[ 52%] Automatic MOC for target logdata
Generating MOC source logdata_autogen/VVDFVJO4II/moc_abstractlogdata.cpp
Generating MOC source logdata_autogen/VVDFVJO4II/moc_logdata.cpp
Generating MOC source logdata_autogen/VVDFVJO4II/moc_logdataworkerthread.cpp
Generating MOC source logdata_autogen/VVDFVJO4II/moc_logfiltereddata.cpp
Generating MOC source logdata_autogen/VVDFVJO4II/moc_logfiltereddataworkerthread.cpp
Generating MOC compilation logdata_autogen/moc_compilation.cpp
[ 52%] Built target logdata_autogen
Scanning dependencies of target logdata
[ 53%] Building CXX object src/logdata/CMakeFiles/logdata.dir/src/abstractlogdata.cpp.obj
abstractlogdata.cpp
[ 54%] Building CXX object src/logdata/CMakeFiles/logdata.dir/src/blockpool.cpp.obj
blockpool.cpp
[ 54%] Building CXX object src/logdata/CMakeFiles/logdata.dir/src/compressedlinestorage.cpp.obj
compressedlinestorage.cpp
[ 55%] Building CXX object src/logdata/CMakeFiles/logdata.dir/src/encodingdetector.cpp.obj
encodingdetector.cpp
[ 56%] Building CXX object src/logdata/CMakeFiles/logdata.dir/src/logdata.cpp.obj
logdata.cpp
[ 56%] Building CXX object src/logdata/CMakeFiles/logdata.dir/src/logdataworkerthread.cpp.obj
logdataworkerthread.cpp
[ 57%] Building CXX object src/logdata/CMakeFiles/logdata.dir/src/logfiltereddata.cpp.obj
logfiltereddata.cpp
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2417): error C2678: binary '*': no operator found which takes a left-hand operand of type 'const Marks::const_iterator' (or there is no acceptable conversion)
D:\glogg\src\logdata\include\data\marks.h(85): note: could be 'const Mark &Marks::const_iterator::operator *(void)'
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2417): note: while trying to match the argument list '(const Marks::const_iterator)'
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2439): note: see reference to function template instantiation '_FwdIt std::_Lower_bound_unchecked<_Iter,_Ty,_Fn>(_FwdIt,_FwdIt,const _Ty &,_Pr)' being compiled
        with
        [
            _FwdIt=Marks::const_iterator,
            _Iter=Marks::const_iterator,
            _Ty=LineNumber,
            _Fn=std::less<void>,
            _Pr=std::less<void>
        ]
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2447): note: see reference to function template instantiation '_FwdIt std::lower_bound<_FwdIt,_Ty,std::less<void>>(_FwdIt,_FwdIt,const _Ty &,_Pr)' being compiled
        with
        [
            _FwdIt=Marks::const_iterator,
            _Ty=LineNumber,
            _Pr=std::less<void>
        ]
d:\glogg\src\logdata\include\data\linetypes.h(177): note: see reference to function template instantiation '_FwdIt std::lower_bound<Iterator,LineNumber>(_FwdIt,_FwdIt,const _Ty &)' being compiled
        with
        [
            _FwdIt=Marks::const_iterator,
            Iterator=Marks::const_iterator,
            _Ty=LineNumber
        ]
D:\glogg\src\logdata\src\logfiltereddata.cpp(416): note: see reference to function template instantiation 'LineNumber lookupLineNumber<Marks::const_iterator>(Iterator,Iterator,LineNumber)' being compiled
        with
        [
            Iterator=Marks::const_iterator
        ]
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2417): error C2100: illegal indirection
NMAKE : fatal error U1077: 'F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\bin\Hostx64\x64\cl.exe' : return code '0x2'
Stop.
NMAKE : fatal error U1077: 'F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\bin\HostX64\x64\nmake.exe' : return code '0x2'
Stop.
NMAKE : fatal error U1077: 'F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\bin\HostX64\x64\nmake.exe' : return code '0x2'
Stop.

D:\glogg\build>
```
</p>
</details><br>

___
Changing `"F:\VisualStudio2017\VC\Auxiliary\Build\vcvars64.bat"` to `"F:\VisualStudio2017\Common7\Tools\VsDevCmd.bat" -arch=x64`, the error still the same, but it is showed a little different:
<details>
<p>

```
D:\glogg\build>"F:\VisualStudio2017\Common7\Tools\VsDevCmd.bat" -arch=x64
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.5.3
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************

D:\glogg\build>"D:\User\Documents\CMake\bin\cmake.exe" --build . --config Release
Git branch: master-variar
Git commit: d066dd8
Git commit: 17.12.0-2-gd066dd8
[  0%] Built target generate_version
[  1%] Built target html_readme
[  1%] Automatic MOC for target uchardet
[  1%] Built target uchardet_autogen
[ 26%] Built target uchardet
[ 27%] Automatic MOC for target efsw
[ 27%] Built target efsw_autogen
[ 44%] Built target efsw
[ 45%] Automatic MOC for target utils
[ 45%] Built target utils_autogen
[ 48%] Built target utils
[ 49%] Automatic MOC for target filewatch
[ 49%] Built target filewatch_autogen
[ 51%] Built target filewatch
[ 52%] Automatic MOC for target logdata
[ 52%] Built target logdata_autogen
[ 53%] Building CXX object src/logdata/CMakeFiles/logdata.dir/src/logfiltereddata.cpp.obj
logfiltereddata.cpp
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2417): error C2678: binary '*': no operator found which takes a left-hand operand of type 'const Marks::const_iterator' (or there is no acceptable conversion)
D:\glogg\src\logdata\include\data\marks.h(85): note: could be 'const Mark &Marks::const_iterator::operator *(void)'
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2417): note: while trying to match the argument list '(const Marks::const_iterator)'
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2439): note: see reference to function template instantiation '_FwdIt std::_Lower_bound_unchecked<_Iter,_Ty,_Fn>(_FwdIt,_FwdIt,const _Ty &,_Pr)' being compiled
        with
        [
            _FwdIt=Marks::const_iterator,
            _Iter=Marks::const_iterator,
            _Ty=LineNumber,
            _Fn=std::less<void>,
            _Pr=std::less<void>
        ]
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2447): note: see reference to function template instantiation '_FwdIt std::lower_bound<_FwdIt,_Ty,std::less<void>>(_FwdIt,_FwdIt,const _Ty &,_Pr)' being compiled
        with
        [
            _FwdIt=Marks::const_iterator,
            _Ty=LineNumber,
            _Pr=std::less<void>
        ]
d:\glogg\src\logdata\include\data\linetypes.h(177): note: see reference to function template instantiation '_FwdIt std::lower_bound<Iterator,LineNumber>(_FwdIt,_FwdIt,const _Ty &)' being compiled
        with
        [
            _FwdIt=Marks::const_iterator,
            Iterator=Marks::const_iterator,
            _Ty=LineNumber
        ]
D:\glogg\src\logdata\src\logfiltereddata.cpp(416): note: see reference to function template instantiation 'LineNumber lookupLineNumber<Marks::const_iterator>(Iterator,Iterator,LineNumber)' being compiled
        with
        [
            Iterator=Marks::const_iterator
        ]
F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\include\algorithm(2417): error C2100: illegal indirection
NMAKE : fatal error U1077: 'F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\bin\Hostx64\x64\cl.exe' : return code '0x2'
Stop.
NMAKE : fatal error U1077: 'F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\bin\HostX64\x64\nmake.exe' : return code '0x2'
Stop.
NMAKE : fatal error U1077: 'F:\VisualStudio2017\VC\Tools\MSVC\14.12.25827\bin\HostX64\x64\nmake.exe' : return code '0x2'
Stop.

D:\glogg\build>
```
</p>
</details><br>

**Comment by variar** [comment]
Thanks, looks like msvc compiler finally got stricter. Last appveyor build used 15.4.4 and ran just fine.

**Comment by variar** [comment]
fixed in f6b211ef69012423a4c1d34ae1bfbea546fbbb30
